src/snowflake/snowpark/session.py

```diff
@@ -2749,4 +2750,34 @@ def _explain_query(self, query: str) -> Optional[str]:
             _logger.warning("query `%s` cannot be explained", query)
             return None
 
+    @staticmethod
+    def connect() -> "Session":
+        """Gets the existing session if one exists (in case of SPs/Streamlits) or creates a new one using default connection params (e.g., in case of SPCS)."""
+
+        # 1) check if in SiS or sproc
+        if (
+            is_in_stored_procedure()
+        ):  # (will be removed before merging) no need for `"streamlit" in sys.modules` check since is_in_stored_procedure() must be true for SiS as well
+            # Basically noop and return the existing session
+            return _get_active_session()
+        elif is_in_spcs():
+            # 2) check if we are running in SPCS
+            with open("/snowflake/session/token") as file:
+                token = file.read()
+            connection_parameters = {
+                "account": os.getenv("SNOWFLAKE_ACCOUNT"),
+                "host": os.getenv("SNOWFLAKE_HOST"),
+                "port": os.getenv("SNOWFLAKE_PORT"),
+                "protocol": os.getenv("SNOWFLAKE_PROTOCOL"),
+                "authenticator": "oauth",
+                "token": token,
+                "warehouse": os.getenv("SNOWFLAKE_WAREHOUSE"),
```

Comment:
SNOWFLAKE_WAREHOUSE may or may not be set. Setting the QUERY_WAREHOUSE on the SERVICE is optional.